File: backend/src/main/java/com/minidocto/model/TimeSlot.java
package com.minidocto.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "time_slots")
public class TimeSlot {
    @Id
    private String id;
    
    private String professionalId;
    
    private LocalDateTime startTime;
    
    private LocalDateTime endTime;
    
    private boolean available = true;
    
    private LocalDateTime createdAt;
}

File: patient_app/lib/models/timeslot.dart
class TimeSlot {
  final String id;
  final String professionalId;
  final DateTime startTime;
  final DateTime endTime;
  final bool available;

  TimeSlot({
    required this.id,
    required this.professionalId,
    required this.startTime,
    required this.endTime,
    required this.available,
  });

  factory TimeSlot.fromJson(Map<String, dynamic> json) {
    return TimeSlot(
      id: json['id'] ?? '',
      professionalId: json['professionalId'] ?? '',
      startTime: DateTime.parse(json['startTime']),
      endTime: DateTime.parse(json['endTime']),
      available: json['available'] ?? false,
    );
  }
}
